### DISCUSSION TITLE
Flaky Tests With Jasmine

### DISCUSSION THREAD

**Discussion by samuelfreiberg** [webdriverio/webdriverio#7526] [Q&A]
In the docs regarding [Retry Flaky Tests](https://webdriver.io/docs/retry/), it mentions that you can add a 4th parameter to to the `it()` function of Jasmine that lets you choose the number of reruns for the test. However, I'm only seeing 3 parameters from my end (the last one being timeout, not retries).

I've also done some digging and haven't found any documentation from Jasmine that leads me to believe there is a 4th parameter as stated in WebDriverIO's documentation.

Has anyone encountered this problem before? If Jasmine doesn't allow this, can we update WebDriverIO's documentation?

**Comment by christian-bromann** [comment]
> However, I'm only seeing 3 parameters from my end (the last one being timeout, not retries).

Can you better describe where do you see only 3 parameters? Note that this is specific to WebdriverIO, you only can use this when using the WebdriverIO testrunner.

**Reply by vrockai** [reply]
And to make it super clear for the OP, here's the line:
https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-jasmine-framework/src/index.ts#L395

**Reply by samuelfreiberg** [reply]
Thanks for the reply! I'm using it() within my .spec file that WDIO runs. Is that what you mean by using the WebDriverIO testrunner?
IE. Using `it()` in test.spec.ts, and in WDIO.conf.js, having my `specs: [test.spec.ts]`

When I CTRL+Click on `it()`, I find the definition is 
```
/**
 * Define a single spec. A spec should contain one or more expectations that test the state of the code.
 * A spec whose expectations all succeed will be passing and a spec with any failures will fail.
 * @param expectation Textual description of what this spec is checking
 * @param assertion Function that contains the code of your test. If not provided the test will be pending.
 * @param timeout Custom timeout for an async spec.
 */
declare function it(expectation: string, assertion?: jasmine.ImplementationCallback, timeout?: number): void;
```
